sklearn/model_selection/_validation.py

```diff
@@ -545,8 +545,9 @@ def permutation_test_score(estimator, X, y, groups=None, cv=None,
         supervised learning.
 
     groups : array-like, with shape (n_samples,), optional
-        Group labels used to constrain the permutation to specific subsets of
-        data.
+        Labels to constrain permutation within groups, i.e. ``y`` values
+        are permuted among samples with the same group identifier.
+        When not specified, ``y`` values are permuted among all samples.
```

Comment:
I felt the usage of GroupCV in `permutation_test_score` to be not a very common usecase... Would you rather have the NOTE heading removed?